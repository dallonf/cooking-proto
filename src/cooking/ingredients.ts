import { makeIngredient, Ingredient } from '../types';

export const rockSalt = makeIngredient({
  name: 'Rock Salt',
  hearts: 1,
});
export const octorokTentacle = makeIngredient({
  name: 'Octorok Tentacle',
  hearts: 5,
  cuisine: 'zoran',
  foodTypes: 'protein'
});
export const duck = makeIngredient({
  name: 'Duck',
  hearts: 5,
  cuisine: 'zoran',
  foodTypes: 'protein',
});
export const bread = makeIngredient({
  name: 'Bread',
  hearts: 2,
  cuisine: 'hylian',
  foodTypes: 'carb',
});
export const buttermilk = makeIngredient({
  name: 'Buttermilk',
  hearts: 1,
  cuisine: 'hylian',
  foodTypes: 'dairy',
  flavorProfiles: 'acidic',
});
export const creme = makeIngredient({
  name: 'Cr√©me',
  hearts: 2,
  cuisine: 'rito',
  foodTypes: 'dairy',
  flavorProfiles: 'sweet',
});
export const artichoke = makeIngredient({
  name: 'Artichoke',
  hearts: 4,
  cuisine: 'rito',
  foodTypes: 'produce',
  flavorProfiles: 'tart'
});
export const tortilla = makeIngredient({
  name: 'Tortilla',
  hearts: 2,
  cuisine: 'gerudo',
  foodTypes: 'carb',
});
export const pepper = makeIngredient({
  name: 'Hot Pepper',
  hearts: 4,
  cuisine: 'gerudo',
  foodTypes: 'produce',
  flavorProfiles: 'spicy',
});

interface IngredientInList extends Readonly<Ingredient> {
  key: string;
}

function makeList(ingredients: { [key: string]: Readonly<Ingredient> }): IngredientInList[] {
  return Object.keys(ingredients).map(key => ({...ingredients[key], key}));
}
export default makeList({
  rockSalt,
  octorokTentacle,
  duck,
  bread,
  buttermilk,
  creme,
  artichoke,
  tortilla,
  pepper,
});