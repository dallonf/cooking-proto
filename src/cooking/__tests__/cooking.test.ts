import cook from '../cooking';
import * as ingredients from '../ingredients';

it('should cook a meal by summing the hearts of its ingredients', function() {
  const meal = cook([ ingredients.rockSalt, ingredients.duck ]);
  expect(meal.hearts).toBe(ingredients.rockSalt.hearts + ingredients.duck.hearts);
});

describe('snapshots', function() {
  it('fajita', function() {
    const meal = cook([ ingredients.tortilla, ingredients.pepper, ingredients.artichoke ]);
    expect(meal).toMatchSnapshot();
  });
});