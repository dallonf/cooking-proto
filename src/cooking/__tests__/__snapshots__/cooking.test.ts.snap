exports[`snapshots butterchoke 1`] = `
Object {
  "effects": Array [
    Object {
      "buffType": "defenseUp",
      "duration": 120,
      "level": 2,
      "type": "buff",
    },
    Object {
      "buffType": "attackUp",
      "duration": 20,
      "level": 1,
      "type": "buff",
    },
  ],
  "hearts": 5,
  "ingredients": Array [
    "artichoke",
    "buttermilk",
  ],
  "name": "A Meal",
}
`;

exports[`snapshots dubious cream 1`] = `
Object {
  "effects": Array [],
  "hearts": -5,
  "ingredients": Array [
    "creme",
    "buttermilk",
  ],
  "name": "Dubious Food",
}
`;

exports[`snapshots mushroom and duck fajita 1`] = `
Object {
  "effects": Array [
    Object {
      "buffType": "stealth",
      "duration": 340,
      "level": -2,
      "type": "buff",
    },
    Object {
      "buffType": "coldResist",
      "duration": 24,
      "level": 1,
      "type": "buff",
    },
  ],
  "hearts": 14,
  "ingredients": Array [
    "tortilla",
    "pepper",
    "duck",
    "silentShroom",
  ],
  "name": "A Meal",
}
`;

exports[`snapshots rito/zora fusion 1`] = `
Object {
  "effects": Array [
    Object {
      "buffType": "defenseUp",
      "duration": 120,
      "level": 2,
      "type": "buff",
    },
    Object {
      "buffType": "stealth",
      "duration": 50,
      "level": 1,
      "type": "buff",
    },
  ],
  "hearts": 12,
  "ingredients": Array [
    "octorokTentacle",
    "artichoke",
    "silentShroom",
  ],
  "name": "A Meal",
}
`;

exports[`snapshots roast duck 1`] = `
Object {
  "effects": Array [],
  "hearts": 6,
  "ingredients": Array [
    "duck",
    "rockSalt",
  ],
  "name": "A Meal",
}
`;

exports[`snapshots veggie fajita 1`] = `
Object {
  "effects": Array [
    Object {
      "buffType": "stealth",
      "duration": 90,
      "level": 2,
      "type": "buff",
    },
    Object {
      "buffType": "coldResist",
      "duration": 24,
      "level": 1,
      "type": "buff",
    },
    Object {
      "buffType": "defenseUp",
      "duration": 120,
      "level": 2,
      "type": "buff",
    },
  ],
  "hearts": 10,
  "ingredients": Array [
    "tortilla",
    "pepper",
    "artichoke",
  ],
  "name": "A Meal",
}
`;
